File: cs50/week6/problem-set/mario/more/mario.py
from cs50 import get_int

while True:
    h = get_int("Enter a unit height for the pyramid: ")
    if (h >= 1 and h <= 8):
        break

# Establishes number of lines to print to representing height
for i in range(h):
    
    # Right-aligned pyramid
    # Prints spaces on each line to right-align first pyramid
    for j in range((h - i) - 1):
        print(' ', end='')
    
    # Prints '#' on each line to construct right-aligned pyramid
    for k in range(i + 1):
        print('#', end='')

    # Prints two spaces to prepare constructing left-aligned pyramid
    print('  ', end='')
    
    # Left-aligned pyramid
    # Prints spaces on each line to right-align the pyramid
    for k in range(i + 1):
        print('#', end='')
    
    # Prints new line to prepare constructing next line of pyramids
    print()

File: cs50/week1/problem-set/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Variable declaration
    int h, i, j, k;
    // Height input from user
    do
    {
        h = get_int("Enter a unit height for the pyramid: ");
    }
    while (h < 1 || h > 8);
    
    // Establishing number of lines
    for (i = 1; i <= h; i++)
    {
    // Left pyramid
        // Print spaces to right-align
        for (k = 1; k <= h - i; k++)
        {
            printf(" ");
        }
        
        // Print '#' on each line 
        for (j = 1; j <= i; j++)
        {
            printf("#");
        }
        // Print two spaces
        printf("  ");
        
    // Right pyramid
       // Print '#' on each line 
        for (j = 1; j <= i; j++)
        {
            printf("#");
        }
        // Print new line
        printf("\n");
    }
}
